pkg/discovery/dns/miekgdns/resolver.go

```diff
@@ -20,6 +20,14 @@ type Resolver struct {
 }
 
 func (r *Resolver) LookupSRV(ctx context.Context, service, proto, name string) (cname string, addrs []*net.SRV, err error) {
+	return r.lookupSRV(service, proto, name, 1, 8)
+}
+
+func (r *Resolver) lookupSRV(service, proto, name string, currIteration, maxIterations int) (cname string, addrs []*net.SRV, err error) {
```

Comment:
I know we use named returns in the exported method, but comparing with `LookupIPAddr`, it would be nice to keep this uniform (my preference would be to not name the returns).

Actually, looking through the code, we never seem to use the `cname` return in any function :thinking:. We might want to simply this and return only `[]*net.SRV, error`. We might need to adjust the interface as well.

Reply:
> I know we use named returns in the exported method, but comparing with LookupIPAddr, it would be nice to keep this uniform (my preference would be to not name the returns).

👍

> Actually, looking through the code, we never seem to use the cname return from LookupIPAddr in any function 🤔. 

yeah I came across this too, but thought it might be some conventional/legacy reasons.

> We might want to simply this and return only []*net.SRV, error.

 Will make the changes :)

> We might need to adjust the interface as well.

sounds like a new Issue to work on :D

Reply:
one more thing worth looking at: https://github.com/thanos-io/thanos/blob/main/pkg/discovery/dns/resolver.go#L38-L39

Reply:
If we remove CNAME as returning parameter, we get an InvalidFaceAssign error in [dns/provider.go](https://github.com/thanos-io/thanos/blob/main/pkg/discovery/dns/provider.go#L49):

`cannot use &(miekgdns.Resolver literal) (value of type *miekgdns.Resolver) as ipLookupResolver value in assignment: *miekgdns.Resolver does not implement ipLookupResolver (wrong type for method LookupSRV)
		have LookupSRV(ctx context.Context, service string, proto string, name string) ([]*net.SRV, error)
		want LookupSRV(ctx context.Context, service string, proto string, name string) (cname string, addrs []*net.SRV, err error)compiler[InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#InvalidIfaceAssign)`

We can change the return parameters to unnamed parameters instead of deleting CNAME. And therefore return `string, []*net.SRV, error`

Any inputs?

Reply:
If we remove CNAME as returning parameter in LookupSRV function, we get an InvalidFaceAssign error in [dns/provider.go](https://github.com/thanos-io/thanos/blob/main/pkg/discovery/dns/provider.go#L49):

`cannot use &(miekgdns.Resolver literal) (value of type *miekgdns.Resolver) as ipLookupResolver value in assignment: *miekgdns.Resolver does not implement ipLookupResolver (wrong type for method LookupSRV)
		have LookupSRV(ctx context.Context, service string, proto string, name string) ([]*net.SRV, error)
		want LookupSRV(ctx context.Context, service string, proto string, name string) (cname string, addrs []*net.SRV, err error)compiler[InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#InvalidIfaceAssign)`

We can change the return parameters to unnamed parameters instead of deleting CNAME. And therefore return `string, []*net.SRV, error`

Any inputs?

Reply:
So I believe this is because of the mentioned interface - since we don't use the `cname` return, my thought was to remove it from the interface completely, i.e. the signature in the interface definition should be changed `LookupSRV(ctx context.Context, service string, proto string, name string) (addrs []*net.SRV, err error)`.

Reply:
I tried changing `ipLookupResolver` interface by removing [cname from LookupSRV function](https://github.com/thanos-io/thanos/blob/main/pkg/discovery/dns/resolver.go#L39) and also changed the LookupSRV function from `miekgdns/resolver.go` but am getting the same error nevertheless

Am I looking at something else?

Reply:
I tried changing `ipLookupResolver` interface by removing [cname from LookupSRV function](https://github.com/thanos-io/thanos/blob/main/pkg/discovery/dns/resolver.go#L39) to `LookupSRV(ctx context.Context, service string, proto string, name string) (addrs []*net.SRV, err error)` and also changed the LookupSRV function from `miekgdns/resolver.go` but am getting the same error nevertheless

Am I looking at something else?

Reply:
Hm, would you mind pushing those changes or creating another branch that you could link to? It would be easier to understand if I could see the code.

Reply:
@matej-g https://github.com/Atharva-Shinde/thanos/tree/local-repo

Reply:
@matej-g https://github.com/Atharva-Shinde/thanos/tree/local-repo
I've created another branch with the changes suggested to look into it

Reply:
Hey @Atharva-Shinde, thanks for this. I now noticed that the other resolver we have in `godns` package actually is wrapped resolver from standard `net` library, so we cannot easily change this :disappointed:. This probably explains why there is that extra `cname` parameter, even though it's unused.

In that case we can just leave the interface as is, sorry for the confusion!